tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchTemplateSearchTests.java

```diff
@@ -0,0 +1,257 @@
+        searchQueryService.expectNodeRefsFromQuery(request, testUser,
+            fileWithTermInName.getNodeRef(), fileWithTermInDescription.getNodeRef(), fileWithTermInTitle.getNodeRef());
+
+        String queryIncludingTag = "TYPE:'cm:content' AND _NODET:" + SEARCH_TERM;
+        SearchRequest requestIncludingTag = req("afts", queryIncludingTag, templates);
+        searchQueryService.expectNodeRefsFromQuery(requestIncludingTag, testUser,
+            fileWithTermInName.getNodeRef(), fileWithTermInDescription.getNodeRef(), fileWithTermInTitle.getNodeRef(), fileWithTermInTag.getNodeRef());
+    }
+
+    @Test(groups = { TestGroup.SEARCH })
+    public void testAftsQuery_templateNameAsQueryDefaultFieldName()
+    {
+        Map<String, String> templates = Map.of("_NODE", "%(cm:name cm:title)");
+        String query = "TYPE:'cm:content' AND " + SEARCH_TERM;
+        SearchRequest request = req("afts", query, templates);
+        request.setDefaults(RestRequestDefaultsModel.builder().defaultFieldName("_NODE").create());
+
+        searchQueryService.expectNodeRefsFromQuery(request, testUser, fileWithTermInName.getNodeRef(), fileWithTermInTitle.getNodeRef());
+    }
+
+    @Test(groups = { TestGroup.SEARCH })
+    public void testAftsQuery_templateWithFixedValue()
+    {
+        Map<String, String> templates = Map.of("_NODE", "%cm:name AND TYPE:'cm:folder'");
+        String query = "_NODE:" + SEARCH_TERM;
+        SearchRequest request = req("afts", query, templates);
+
+        searchQueryService.expectNodeRefsFromQuery(request, testUser, folderWithTermInName.getNodeRef());
+    }
+
+    @Test(groups = { TestGroup.SEARCH })
+    public void testAftsQuery_boostedTemplate()
+    {
+        Map<String, String> templates = Map.of("_NODE", "%cm:name");
+        String query = "TYPE:'cm:content' AND _NODE:" + SEARCH_TERM + "^0.5 OR _NODE:dummy^2";
+        SearchRequest request = req("afts", query, templates);
+        searchQueryService.expectResultsInOrder(request, testUser, true, fileWithDifferentTermInName.getName(), fileWithTermInName.getName());
+
+        String queryInvertedBoost = "TYPE:'cm:content' AND _NODE:" + SEARCH_TERM + "^2 OR _NODE:dummy^0.5";
+        SearchRequest requestInvertedBoost = req("afts", queryInvertedBoost, templates);
+        searchQueryService.expectResultsInOrder(requestInvertedBoost, testUser, false, fileWithTermInName.getName(), fileWithDifferentTermInName.getName());
+    }
+
+    @Test(groups = { TestGroup.SEARCH })
+    public void testAftsQuery_expandedTemplate()
+    {
+        Map<String, String> templates = Map.of("_NODE", "%cm:name");
+        String query = "TYPE:'cm:content' AND ~_NODE:" + SEARCH_TERM;
+        SearchRequest request = req("afts", query, templates);
+
+        searchQueryService.expectNodeRefsFromQuery(request, testUser, fileWithTermInName.getNodeRef());
+    }
+
+    /* ********************* Helper methods ********************* */
+
```

Comment:
```suggestion

```
I think it's clear enough that anything that is not annotated with `@Test` is most likely an helper method, I'd avoid this kind of marker comments in the code as they're easy to miss and end up with Tests below it.